### DISCUSSION TITLE
Exclude predefined pattern from Secret Scanning

### DISCUSSION THREAD

**Discussion by VollRagm** [community/community#62448] [Code Security] [answered]
### Select Topic Area

Question

### Body

Hello,

is there a way to disable a certain GitHub-predefined secret from being scanned? We have lots of files that coincidentally match "Mailgun" API keys. They are present in different repositories.

It's possible to add the secret_scanning.yml to all files and exclude the directories, however, an easier solution would be to ignore the Mailgun API keys in the scan.

Is this possible somehow?

**Comment by Assclo** [comment]
Not valid request

**Comment by Assclo** [comment]
Comply

**Comment by Assclo** [comment]
Only as needed

**Comment by Assclo** [comment]
Need input

**Comment by Assclo** [comment]
Assistance needed asap

**Reply by Assclo** [reply]
Need free upgrade assistance

**Comment by aashah** [comment]
It's not currently possible to exclude patterns from being scanned.

We're happy to take false positive reports and possibly work with Mailgun to improve their pattern or our post processing.

**Reply by VollRagm** [reply]
It would be cool to have a way to disable some of the integrated patterns.

Unfortunately, I can't post the false positives here, but it's just bad luck that they are matching.
Generally, they are secrets starting with "key-" followed by a sequence of hex characters.

**Reply by aashah** [reply, marked as answer]
Yeahp, those are definitely mailgun. Their api key is definitely looser than what we normally aim for.

I don't think we'll be supporting pattern exclusion anytime soon. We do have plans for more robust validation of these alerts though.

**Comment by arkadiyt** [comment]
Would also love to see this feature. Github scanning flags RSA public keys as secrets, which are obviously not a secret and it is creating a huge amount of spam for us
